analytics_data_api/v0/tests/views/test_learners.py

```diff
@@ -404,9 +412,192 @@ def test_pagination(self):
         ({'course_id': 'edX/DemoX/Demo_Course', 'page': 'bad_value'}, 'illegal_parameter_values'),
         ({'course_id': 'edX/DemoX/Demo_Course', 'page_size': 'bad_value'}, 'illegal_parameter_values'),
         ({'course_id': 'edX/DemoX/Demo_Course', 'page_size': 101}, 'illegal_parameter_values'),
+        ({'course_id': 'edX/DemoX/Demo_Course', 'segments': 'a_non_existent_segment'}, 'illegal_parameter_values'),
+        ({'course_id': 'edX/DemoX/Demo_Course', 'ignore_segments': 'a_non_existent_segment'},
+         'illegal_parameter_values'),
     )
     @ddt.unpack
     def test_bad_request(self, parameters, expected_error_code):
         response = self.authenticated_get('/api/v0/learners/', parameters)
         self.assertEqual(response.status_code, 400)
         self.assertEqual(json.loads(response.content)['error_code'], expected_error_code)
+
+
+@ddt.ddt
+class CourseLearnerMetadataTests(DemoCourseMixin, VerifyCourseIdMixin,
```

Comment:
I had to remind myself why we're calling the mixins first in the MRO (usually they're called after the parent class), which I thought was because of this bug - https://code.djangoproject.com/ticket/15828, but it looks like it was fixed a while ago. Does this work when the mixins come after the `TestCaseWithAuthentication`?